src/server/pkg/stream/priority_queue.go

```diff
@@ -0,0 +1,136 @@
+package stream
```

Comment:
stream seems like an undescriptive name for this package. Can we call it priority queue instead?

Reply:
I did not create this package with the intention of only having the priority queue. It was intended for abstractions that operate on streams, such as the priority queue. Stream is kind of a generic name, but the idea would be that we would have a standard stream interface that we can perform operations on (the stream interface could be improved, but it exists as is for legacy reasons). I'm fine with renaming it (priority queue does not seem like an appropriate name because it is not a general purpose priority queue, it only operates on streams as defined by the stream interface), but that was the idea.